### ISSUE TITLE
Support ignoring dependencies per language

### ISSUE THREAD

**Issue by AlistairB** [deadpendency/deadpendency#26]
## Why?

#22 breaks down the report by programming language, but currently it is tricky to break down the ignore section by programming language. Deadpendency could look at what dep file the ignored deps come from, then use that information. However, really ignored deps should already be specific to a language. For example you may want to ignore a dep `request` in javascript, but then use a dep `request` in Haskell. Currently you either ignore for all languages, or allow for all.

## Acceptance Criteria

- Support the following:

```yaml
ignore-failures:
  javascript:
    - this-is-js-dep
    - other-languages-can-use-this-dep-name
```

- global ignores will no longer be supported. There is no reason to want to ignore the same dep name in many languages.

**Comment by AlistairB** [comment]
This issue pokes at some minor flaws in the current modelling. I'd like to resolve these all together (or before doing this issue).

## Deduplicating dependencies is per language

If we have a `JavaScript` dependency of `blah` and a `Haskell` dependency of `blah`, they are not considered duplicates and both will be fetched. This is good.

It get's more complex when you have duplicates of `JavaScript` and `TypeScript`, or Java/Kotlin or C# / VB.NET where the registry is the same. These will resolve to the same dependency report, so should be considered duplicates.

## Ignore per registry

Similarly, ignoring would make sense at the registry level, not the language level.

**Comment by AlistairB** [comment]
As from `0.3.9.0` deduplicating is now per registry not programming language.

**Comment by AlistairB** [comment]
`0.3.10.0` updates to support ignoring deps per language. Global ignores are working for now (undocumented feature), but will be removed in the future once a good deprecation plan is in place.
